bug/4735-event-file-is-bigger-after-updating-event-creation

briemla opened the pull request:
OP#4735

randy-seng reviewed (commented):
The event creation has slowed down, I added some changes to speed up removing duplicate events from the event repository. It should be faster now. The benchmark is also running as of writing this comment. @briemla Can you review my changes. I'll approve this PR when I get your ok.

briemla commented:
> The event creation has slowed down, I added some changes to speed up removing duplicate events from the event repository. It should be faster now. The benchmark is also running as of writing this comment. @briemla Can you review my changes. I'll approve this PR when I get your ok.

Looks good to me. I switched the invocation order in `add`. Duplicates will now be discarded before sorting as it already was done in `add_all`. This is to be consistent between calls.
